Formula/lua@5.3.rb

```diff
@@ -0,0 +1,169 @@
+class LuaAT53 < Formula
+  desc "Powerful, lightweight programming language"
+  homepage "https://www.lua.org/"
+  url "https://www.lua.org/ftp/lua-5.3.3.tar.gz"
+  sha256 "5113c06884f7de453ce57702abaac1d618307f33f6789fa870e87a59d772aca2"
+
+  option "with-default-names", "Don't version-suffix the Lua installation. Conflicts with Homebrew's Lua"
+  option "without-luarocks", "Don't build with Luarocks support embedded"
+
+  # Be sure to build a dylib, or else runtime modules will pull in another static copy of liblua = crashy
+  # See: https://github.com/Homebrew/homebrew/pull/5043
+  # ***Update me with each version bump!***
+  patch :DATA
```

Comment:
@MikeMcQuaid This patch hack can be found in formulae for lua 5.1, 5.2 & 5.3. Perhaps let's just leave this for a future pull request?